app/Http/Controllers/UserController.php

```diff
@@ -30,14 +32,33 @@ public function store(RegisterRequest $request)
     public function login(LoginRequest $request)
     {
         $requestData = $request->only(['email', 'password']);
-        if(!Auth::attempt($requestData))
-        {
+
+        if(!Auth::attempt($requestData)) {
 
             return response(['message' => 'Bad data'], 401);
         }
+
         $token = Auth::user()->createToken('AuthToken')->accessToken;
         $response = ['token' => $token];
 
         return response($response, 200);
     }
+    public function update(UpdateUserRequest $request)
+    {
+        $user = User::where('id', $request->id)->first();
+
+        if (!$user) {
+            return response('error', 404);
+        }
+        if (Auth::user()->can('update', $user)) {
+            $user = $this->userService->updateUser($request->id, $request->name, $request->email);
```

Comment:
What if the user will have more than 2 properties? You should pass an array of data for ex: int $id, array $request->validated()
and use $user->fill($data)